doc/modules/preprocessing.rst

```diff
@@ -539,9 +554,71 @@ columns for this feature will be all zeros
     >>> enc.transform([['female', 'from Asia', 'uses Chrome']]).toarray()
     array([[1., 0., 0., 0., 0., 0.]])
 
+See :ref:`dict_feature_extraction` for categorical features that are
+represented as a dict, not as scalars.
+
+.. _unary_encoding:
+
+Unary encoding
+--------------
+
+For some ordinal features, it does not necessarily make sense to use
+:class:`OrdinalEncoder` if the difference between the ordered categories is
+uneven, for example with a feature that takes values in "very short",
+"short", "big".
+
+For such features, it is possible to use a unary encoding, which is
+implemented in :class:`UnaryEncoder`. This encoder transforms each ordinal
+feature with ``m`` possible values into ``m - 1`` binary features, where the
+ith feature is active if x > i. For example::
+
+  >>> enc = preprocessing.UnaryEncoder()
+  >>> enc.fit([[0, 0, 3], [1, 1, 0], [0, 2, 1], [1, 0, 2]]) # doctest: +ELLIPSIS
+  UnaryEncoder(dtype=<... 'numpy.float64'>, handle_greater='warn',
+               max_value='auto', sparse=False)
+  >>> enc.transform([[0, 1, 3]])
+  array([[0., 1., 0., 1., 1., 1.]])
+
+Here the first feature with 2 categories is transformed into 1 column, the
+second feature with 3 values is transformed into 2 columns, and the third
+feature is transformed into 3 columns.
+
+By default, the number of categories in a feature is inferred automatically
+from the dataset by looking for the maximum value. It is possible to specify
+this explicitly using the parameter ``max_value``. In particular if the
+training data might have missing categorical features, one has to explicitly
+set ``max_value``. For example,::
+
+  >>> enc = preprocessing.UnaryEncoder(max_value=[2, 3, 4])
+  >>> # Note that there are missing categorical values for the 2nd and 3rd
+  >>> # features
+  >>> enc.fit([[1, 2, 3], [0, 2, 0]])  # doctest: +ELLIPSIS
+  UnaryEncoder(dtype=<... 'numpy.float64'>, handle_greater='warn',
+               max_value=[2, 3, 4], sparse=False)
+  >>> enc.transform([[1, 1, 2]])
+  array([[1., 1., 0., 1., 1., 0.]])
+
+.. note::
+
+  This encoding is likely to help when used with linear models and
+  kernel-based models like SVMs with the standard kernels. On the other
+  hand, this transformation is unlikely to help when using tree-based
+  models, since those already work on the basis of a particular feature
+  value being less or bigger than a threshold.
+
+It is possible to combine :class:`UnaryEncoder` and :class:`OrdinalEncoder`
```

Comment:
"In case the input variable is not represented as a number from 0 to max_value, it is possible..."